hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -492,16 +507,82 @@ func oneOfHandler(ctx context.Context, scanner *SchemaScanner, schema *gojsonsch
 		}
 
 		if result != nil {
-			results = append(results, result)
+			results = appendIfUniqueType(results, result)
 		}
 	}
 
 	if len(results) == 1 {
 		return results[0], nil
 	}
 
-	// bail out, can't handle this yet:
-	return astmodel.AnyType, nil
+	// If there's more than one option, synthesize a type.
+	// Note that this is required because Kubernetes CRDs do not support OneOf the same way
+	// OpenAPI does, see https://github.com/Azure/k8s-infra/issues/71
+	var fields []*astmodel.FieldDefinition
+
+	for i, t := range results {
```

Comment:
A thought on naming - the types have their own names (e.g. `Foo`, `Bar` and `Baz` from your samples); can we use those to generate names for the fields?

Reply:
That's what the first case (`TypeName`) is already doing -- using the name of the ref to generate the property. All of the other cases (`EnumType`/`StructType`, etc) are for anonymous types embedded in the OneOf (see the `OneOf_generates_wrapper_type_with_named_properties_from_anonymous_types` golden file test) - there are not really any "good" names to use in this case unfortunately.